clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -308,6 +358,19 @@ void SYCL::fpga::BackendCompiler::ConstructJob(
           getToolChain().getTriple().getArch() == llvm::Triple::spir64) &&
          "Unsupported target");
 
+  // Grab the -Xsycl-target* options.
+  const toolchains::SYCLToolChain &TC =
+      static_cast<const toolchains::SYCLToolChain &>(getToolChain());
+  ArgStringList TargetArgs;
+  TC.TranslateBackendTargetArgs(Args, TargetArgs);
+
+  // When performing emulation compilations for FPGA AOT, we want to use
+  // opencl-aot instead of aoc.
+  if (C.getDriver().isFPGAEmulationMode()) {
```

Comment:
How would we have reached this block if the condition was not true?

Reply:
We call the BE compiler for FPGA for `spir64_fpga` regardless of emulation mode or not.

Reply:
Ah, of course. I realized this on the design review stage, but then it somehow slipped my mind when the Toolchain time came :D